msprime/tables.py

```diff
@@ -1562,21 +1611,40 @@ def sort(self, edge_start=0):
         given parent ID are adjacent, but we do not require that they be listed in
         sorted order.
 
-        Sites are sorted by position.
+        Sites are sorted by position, and sites with the same position retain
+        their relative ordering.
 
-        Mutations are sorted by site ID.
+        Mutations are sorted by site ID, and mutations with the same site retain
+        their relative ordering.
```

Comment:
sort does not ensure that mutations occur after their parents, unless this is already the case, contradicting what was said above: `ensuring that all tree sequence ordering requirements are met.`